posthog/models/property/util.py

```diff
@@ -244,7 +265,11 @@ def parse_prop_clauses(
                 idx,
                 prepend=f"personquery_{prepend}",
                 allow_denormalized_props=True,
-                transform_expression=lambda column_name: f"argMax(person.{column_name}, version)",
+                transform_expression=(
```

Comment:
I don't think you need this new parameter, what you're doing is the difference between the existing two property modes: `DIRECT` and `DIRECT_ON_PERSONS`.

See `_get_grouped_person_filters` and `_get_person_filters` - they're doing the same thing - you want the same thing, but for the search clauses. Would be nice to re-use things here.

Reply:
Hmm, okay, that's useful! I have to say though that `DIRECT` vs `DIRECT_ON_PERSONS` absolutely doesn't suggest to me any specific difference. 😅 And `PersonPropertiesMode` has no documentation or comments. So I would super appreciate something there.

Reply:
Agree, realised this while typing this comment 😅